packages/editor/src/components/ComponentForm/ComponentForm.tsx

```diff
@@ -74,43 +70,74 @@ export const ComponentForm: React.FC<Props> = props => {
   }
   const { version, name } = parseType(selectedComponent.type);
   const cImpl = registry.getComponent(version, name);
-  const properties =  Object.assign(
+  const properties = Object.assign(
     parseTypeBox(cImpl.spec.properties as TSchema),
-    selectedComponent.properties,
+    selectedComponent.properties
   );
-  const fields = Object.keys(properties || []).map(key => {
+  const propertyFields = Object.keys(properties || []).map(key => {
     const value = properties![key];
     return renderField({ key, value, fullKey: key, selectedId });
   });
 
-  const traitForms = selectedComponent.traits.map(t => {
-    const traitForm = Object.keys(t.properties || []).map(key => {
+  const traitFields = selectedComponent.traits.map(t => {
+    if (ignoreTraitsList.includes(t.type)) return null;
+    return Object.keys(t.properties || []).map(key => {
       const value = t.properties[key];
       return renderField({ key, value, fullKey: key, type: t.type, selectedId });
     });
-
-    return (
-      <form key={t.type}>
-        <strong>{t.type}</strong>
-        {traitForm}
-      </form>
-    );
   });
 
+  const eventHandlers = selectedComponent.traits.find(t => t.type === 'core/v1/event')
```

Comment:
Maybe we can pass selectedComponent to `EventTraitForm`, and let it handle this inside the component.

This will help others define their own trait/component form without change the `ComponentForm` code.